webapp/channels/src/components/permissions_gates/team_permission_gate/team_permission_gate.tsx

```diff
@@ -23,26 +23,15 @@ type Props = {
     /**
      * Invert the permission (used for else)
      */
-    invert: boolean;
+    invert?: boolean;
 
     /**
      * Content protected by the permissions gate
      */
     children: React.ReactNode;
 };
 
-export default class TeamPermissionGate extends React.PureComponent<Props> {
-    public static defaultProps = {
-        invert: false,
-    };
-
-    render() {
-        if (this.props.hasPermission && !this.props.invert) {
-            return this.props.children;
-        }
-        if (!this.props.hasPermission && this.props.invert) {
-            return this.props.children;
-        }
-        return null;
-    }
-}
+const TeamPermissionGate = ({invert = false, hasPermission, children}: Props) =>
+    <>{(hasPermission || invert) && children}</>;
```

Comment:
This condition is wrong. The idea is `hasPermission !== invert`. That being said, can you leave it in the if version?

```
if (hasPermission !== invert) {
    return children;
}

return null;
```

Reply:
Sure !